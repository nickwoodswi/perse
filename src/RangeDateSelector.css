.date-selector{
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
}
.month-selector{
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: 50px;
    font-size: 16px;
    margin-right: 5px;
    margin-top: 10px;
    border-radius: 5px;
    font-weight: bold;
    border: none;
    box-shadow: 2px 2px 2px darkgrey;
}
.day-selector{
    display: flex;
    justify-content: center;
    align-items: center;
    width: 48%;
    height: 50px;
    font-size: 16px;
    margin-right: 5px;
    margin-top: 10px;
    border-radius: 5px;
    font-weight: bold;
    border: none;
    box-shadow: 2px 2px 2px darkgrey;
}
.year-selector{
    display: flex;
    justify-content: center;
    align-items: center;
    width: 48%;
    height: 50px;
    font-size: 16px;
    margin-right: 5px;
    margin-top: 10px;
    border-radius: 5px;
    font-weight: bold;
    border: none;
    box-shadow: 2px 2px 2px darkgrey;
}
.range-date-divider{
    display: flex;
    justify-content: center;
    margin-top: 10px;
    padding-top: 5px;
    padding-bottom: 5px;
    width: 98%;
    font-weight: bold;
    color: white;
}
