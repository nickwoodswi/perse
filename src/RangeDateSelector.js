import React, { Component } from 'react'
import MonthDayCounts from './MonthDayCounts'
import './RangeDateSelector.css'

class RangeDateSelector extends Component{
    render() {
        const months = Object.keys(MonthDayCounts)
        const startDays = MonthDayCounts[this.props.selectedStartMonth]
        const endDays = MonthDayCounts[this.props.selectedEndMonth]
        return(
            <>
            <div className="range-date-selector">
                <div className="date-selector">
                    <label htmlFor="start-month-selector">Select Start Month</label>
                    <select id="start-month-selector"
                        key="start-month-selector" 
                        className="month-selector" 
                        onChange={e => this.props.handleRangeDate('month_start', e.target.value)}>
                            {months.map((month, idx) => {
                                return(
                                    <option key={idx}>{month}</option>
                                )
                            })}
                    </select>
                    <label htmlFor="start-day-selector">Select Start Day</label>
                    <select 
                        id="start-day-selector"
                        key="start-day-selector" 
                        className="day-selector" 
                        onChange={e => this.props.handleRangeDate('day_start', e.target.value)}>
                        <>{startDays}</>
                    </select>
                    <label htmlFor="start-year-selector">Select Start Year</label>
                    <select 
                        id="start-year-selector"
                        key="start-year-selector" 
                        className="year-selector" 
                        onChange={e => this.props.handleRangeDate('year_start', e.target.value)}>
                            <option>SELECT YEAR</option>
                            <option>2020</option>
                            <option>2021</option>
                    </select>
                </div>

                <div className="range-date-divider">THRU</div>

                <div className="date-selector">
                    <label htmlFor="end-month-selector">Select End Month</label>
                    <select 
                        id="end-month-selector"
                        key="end-month-selector" 
                        className="month-selector" 
                        onChange={e => this.props.handleRangeDate('month_end', e.target.value)}>
                            {months.map((month, idx) => {
                                return(
                                    <option key={idx}>{month}</option>
                                )
                            })}
                    </select>
                    <label htmlFor="end-day-selector">Select End Day</label>
                    <select 
                        id="end-day-selector"
                        key="end-day-selector" 
                        className="day-selector" 
                        onChange={e => this.props.handleRangeDate('day_end', e.target.value)}>
                        <>{endDays}</>
                    </select>
                    <label htmlFor="end-year-selector">Select End Year</label>
                    <select 
                        id="end-year-selector"
                        key="end-year-selector" 
                        className="year-selector" 
                        onChange={e => this.props.handleRangeDate('year_end', e.target.value)}>
                            <option>2020</option>
                            <option>2021</option>
                    </select>
                </div>
            </div>
            </>
        )
    }
}

export default RangeDateSelector