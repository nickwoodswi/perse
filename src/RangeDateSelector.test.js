import React from 'react';
import { render } from '@testing-library/react';
import RangeDateSelector from './RangeDateSelector';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

test('renders RangeDateSelector component', () => {
    render(<MemoryRouter><RangeDateSelector /></MemoryRouter>);
});

it('RangeDateSelector renders the UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><RangeDateSelector /></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});