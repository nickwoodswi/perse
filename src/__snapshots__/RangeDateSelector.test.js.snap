// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeDateSelector renders the UI as expected 1`] = `
<div
  className="range-date-selector"
>
  <div
    className="date-selector"
  >
    <select
      className="month-selector"
      onChange={[Function]}
    >
      <option>
        MONTH
      </option>
      <option>
        January
      </option>
      <option>
        February
      </option>
      <option>
        March
      </option>
      <option>
        April
      </option>
      <option>
        May
      </option>
      <option>
        June
      </option>
      <option>
        July
      </option>
      <option>
        August
      </option>
      <option>
        September
      </option>
      <option>
        October
      </option>
      <option>
        November
      </option>
      <option>
        December
      </option>
    </select>
    <select
      className="day-selector"
      onChange={[Function]}
    />
    <select
      className="year-selector"
      onChange={[Function]}
    >
      <option>
        2020
      </option>
      <option>
        2021
      </option>
    </select>
  </div>
  <div
    className="range-date-divider"
  >
    THRU
  </div>
  <div
    className="date-selector"
  >
    <select
      className="month-selector"
      onChange={[Function]}
    >
      <option>
        MONTH
      </option>
      <option>
        January
      </option>
      <option>
        February
      </option>
      <option>
        March
      </option>
      <option>
        April
      </option>
      <option>
        May
      </option>
      <option>
        June
      </option>
      <option>
        July
      </option>
      <option>
        August
      </option>
      <option>
        September
      </option>
      <option>
        October
      </option>
      <option>
        November
      </option>
      <option>
        December
      </option>
    </select>
    <select
      className="day-selector"
      onChange={[Function]}
    />
    <select
      className="year-selector"
      onChange={[Function]}
    >
      <option>
        2020
      </option>
      <option>
        2021
      </option>
    </select>
  </div>
</div>
`;
